server/src/test/java/io/crate/integrationtests/DeleteIntegrationTest.java

```diff
@@ -310,7 +310,7 @@ public void test_delete_partitions_from_subquery_does_not_leave_empty_orphan_par
 
     @Test
     public void test_can_reuse_prepared_statement_for_delete_containing_non_deterministic_function() throws Exception {
```

Comment:
imho, we should at least rename the test, or add a dedicated test for a partitioned table. Please check if this can be done in a unit test instead, we should get a `DeletePartitions` plan AFAIK, no?

Reply:
> we should get a DeletePartitions plan

Actually, we get a `Delete` plan and reach `WhereClauseAnalyzer` when `executeOrFail()` is called. I will add a dedicated test(not sure if a unit test can cover this).

Reply:
Yes, please add a dedicated test. The issue has nothing to do with the test you've changed, at least not with the test name.

Reply:
how about unit testing: `resolvePartitions()` ?
Also, are we sure that other non-deterministic functions should work here?

Reply:
not sure, sorry, do you have a list of the non-deterministic functions?

Reply:
I skimmed through https://cratedb.com/docs/crate/reference/en/latest/general/builtins/scalar-functions.html,
```
random
gen_random_text_uuid
current_time current_timestamp
curdate
current_date
now
```

Reply:
I think we're ok then. thx.

Reply:
Thank you, could you please take a quick look at the test?